ascending navigation (#2227)
grommet/hpe-design-system: aries-site/src/pages/templates/ascending-navigation.mdx

import { DiagramPreview, Example } from '../../layouts';

When an application’s hierarchy is well designed, the ascending button will typically take the user 
back to the previous page they came from.

<Example 
  caption="caption TBD" 
  plain 
  pad="small"
  previewWidth="medium"
  height={{ min: 'small' }}
>
  <DiagramPreview 
    src={{
      dark: "/templateImages/ascending-button-preview-invert.png",
      light: '/templateImages/ascending-button-preview.png'
    }} 
    alt="caption TBD" />
</Example>

## Guidance

> - Follows the default button styling with a left arrow icon.

Reviewer: ```suggestion
- Follows the default button styling with a "FormPrevious" icon.
```

Trying to align language with implementation.

> - Follows the default button styling with a "FormPrevious" icon.

Reviewer 2: I know this terminology came up in today's share out. I'm curious if there can be confusion with using "FormPrevious" as the icon name as it isn't remotely close to a form. In Figma, this icon is called "Previous". I'm just wondering if its called the same in the theme? Or is it "FormPrevious"?

Author: This is where it's confusing which icon to use there is an icon that is "Previous" as well as "FormPrevious"

> - Displayed on the left, top corner of the application.

Reviewer: It's not really just the "left, top" corner. It seems like there is more specific guidance we could provide -->

"Place in the page header above any page content and aligned with the left edge of the content." Or something like that (depending on where we land design-wise).

Reviewer 2: This is still WIP. On Wed. I had a chat w/ Aditi on placement since we've been seeing it be placed in different areas of each service team. We both agreed it should be aligned with content. So this will be updated.

Reviewer 2: Displayed below the header and aligned to the left with the page's content.

import { DiagramPreview, Example } from '../../layouts';

When an application’s hierarchy is well designed, the ascending button will typically take the user 
back to the previous page they came from.

<Example 

>   caption="Example of an acending button." 

Reviewer 3: ```suggestion
  caption="Example of an ascending button." 
```

  plain 
  pad="small"
  previewWidth="medium"
  height={{ min: 'small' }}
>

>   <DiagramPreview 

Reviewer 3: Why is there a grey bar at the top of this image? It looks odd and at worst sloppy.

Reviewer 2: I don't see the grey bar you're referring to...

Author: Ill fix this I had the wrong image

    src={{
      dark: "/templateImages/ascending-button-preview-invert.svg",
      light: '/templateImages/ascending-button-preview.svg'
    }} 
    alt="ascending button in an application" />
</Example>

## Guidance
- Follows the [default button styling](https://design-system.hpe.design/components/button#default-button) with a "FormPrevious" icon.
- Displayed on the left, top corner of the application.

## When to use
- To allow users to return to the parent (hub) screen when on a details (child/spoke) page.
- To visually communicate the parent page for the user’s current page.

### Ascending Navigation Example
Ascending Navigation provides an easy way for a user to move from a child page up to a parent page.

> <Example 

Reviewer 2: I would 

> I am concerned that this implementation is one giant image and is inaccessible. How is a user of a screen reader supposed to receive the information (e.g. embedded text)?
> 
> At a minimum, each frame/step should be broken up.

I would agree that it should be broken down into its own frame. We've tried many different options and this seemed to be the best bet.

>   caption="A diagram of a user experience using multiple types of Navigation such as Drill Down to look at a details view of a  row
>   on a table, Matrix Navigation to jump to another page and Ascending Button Navigation to navigate to the parent page." 

Reviewer 3: ```suggestion
  caption="A diagram of a user experience using multiple types of navigation such as Drill Down to look at a details view of a row
  on a table, Matrix Navigation to jump to another page, and Ascending Navigation to navigate to the parent page." 
```
